mountpoint-s3-crt/src/common/string.rs

```diff
@@ -0,0 +1,41 @@
+//! CRT strings are immutable bytes with no inherent encoding. Most CRT APIs use byte buffers
+//! instead, so this binding for `aws_string` is mostly for automatic memory management.
+
+use std::ptr::NonNull;
+
+use mountpoint_s3_crt_sys::*;
+
+use crate::common::allocator::Allocator;
+
+/// An immutable string holding either text or binary data.
+// We call this `AwsString` just because Rust's `String` type is in the prelude and we don't want to
+// confuse the two.
+#[derive(Debug)]
+pub struct AwsString {
+    inner: NonNull<aws_string>,
+}
+
+impl AwsString {
```

Comment:
Can we unit test this and validate we're dropping, etc.?

Reply:
Not easily -- the undering `aws_string` gets deallocated as part of `aws_string_destroy`, and so the `NonNull<aws_string>` becomes a dangling pointer after that point. We could try to peek inside the allocator to check it was actually deallocated, but that would be racy with other threads running at test time since they all share the same allocator.